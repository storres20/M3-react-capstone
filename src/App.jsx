import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  BrowserRouter,
  Routes,
  Route,
} from 'react-router-dom';

import MoviesGrid from './pages/Category/MoviesGrid';
import AboutPage from './pages/AboutPage/AboutPage';
import NotFoundPage from './components/NotFoundPage';
import NavbarRB from './components/Navbar/NavbarRB';
import MovieDetails from './pages/DetailPage/MovieDetails';
import HomePage from './pages/HomePage/HomePage';
import { getCategory } from './redux/movies/categories';

export default function App() {
  /* Load Redux State */
  const category = useSelector((state) => state.category);
  // *********************************

  /* Setup Redux dispatch */
  const dispatch = useDispatch();

  /* Dispatch only once */
  useEffect(() => {
    if (category.length === 0) {
      dispatch(getCategory());
    }
  }, [dispatch, category]);

  return (
    <BrowserRouter>
      <NavbarRB />

      <main>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/category/:catId" element={<MoviesGrid />} />
          <Route path="/about" element={<AboutPage />} />
          <Route path="/detailPage/:movieId" element={<MovieDetails />} />
          <Route path="/*" element={<NotFoundPage />} />
        </Routes>
      </main>
    </BrowserRouter>
  );
}
