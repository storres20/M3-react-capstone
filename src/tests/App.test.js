import React from 'react';
import renderer from 'react-test-renderer';
// Redux
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
// ****
import App from '../App';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
