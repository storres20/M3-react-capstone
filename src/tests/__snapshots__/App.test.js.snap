// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top"
  >
    <div
      className="container"
    >
      <span
        className="NavTitle navbar-brand"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Movies
        </a>
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto NavBar navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            aria-current="page"
            className="NavSubTitle active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="NavSubTitle"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
        <div
          className="designContainer"
        >
          <p>
            Designed by
             
            <a
              href="https://myport20.netlify.app/"
              rel="noreferrer"
              target="_blank"
            >
              <b>
                ITALO
              </b>
            </a>
          </p>
        </div>
      </div>
    </div>
  </nav>,
  <main>
    <div>
      <h2
        className="title"
      >
        Categories
      </h2>
      <ul
        className="moviesGrid"
      />
      <div
        className="button"
      />
    </div>
  </main>,
]
`;
